### ISSUE TITLE
Kontrol Weekly Update

### ISSUE THREAD

**Issue by yale-vinson** [runtimeverification/kontrol#83]

**Comment by yale-vinson** [comment]
### TL;DR Summary

**Where We Are**
- [Functions with array arguments give "Unsupported ABI type for method` error](https://github.com/runtimeverification/kontrol/issues/55) (Hao)
- [Parallelize symbolic execution of tests](https://github.com/runtimeverification/evm-semantics/issues/2009) (Noah)
- [Issues running kclaim style proofs](https://github.com/runtimeverification/evm-semantics/issues/2088) (Noah)
- [Add struct support](https://github.com/runtimeverification/evm-semantics/pull/2051) (Palina)
- [Optimize symbolic execution using depth-based call graphs](https://github.com/runtimeverification/evm-semantics/issues/2066) (Andrei)
- [Clarify and Improve --test Parameter Functionality in kevm foundry-prove](https://github.com/runtimeverification/evm-semantics/issues/2063) (Andrei)
- [Downstream testing using engagement codebases](https://github.com/runtimeverification/kontrol/issues/17) (Juan)
- [kevm foundry-prove: not running the constructor leads to unexpected behavior](https://github.com/runtimeverification/kontrol/issues/66) (Noah)
- [Execution of contract initialization code](https://github.com/runtimeverification/kontrol/pull/54) (Noah)
- [Add '--smt-tactic <s-expression>' command-line option](https://github.com/runtimeverification/evm-semantics/pull/2086) (Georgy)
- [Introduce booster-doctor --- kore-rcp-booster abort analysis tool](https://github.com/runtimeverification/pyk/pull/659) (Georgy)
- [Make booster the default, disable tests on legacy backend](https://github.com/runtimeverification/kontrol/pull/75) (Everett)

**Up Next**
- [Usability improvements to foundry-merge-node](https://github.com/runtimeverification/kontrol/issues/23)
- [Automatically detect and warn user on bad branches](https://github.com/runtimeverification/evm-semantics/issues/1752)
- [Report SMT solver timeout in CLI output](https://github.com/runtimeverification/kontrol/issues/21)
- [Outdated K causes incorrect KEVM/Kontrol output](https://github.com/runtimeverification/evm-semantics/issues/2097)
- [Run Nix tests on CI](https://github.com/runtimeverification/kontrol/issues/76)
- [Move foundry.md from evm-semantics to kontrol](https://github.com/runtimeverification/kontrol/issues/61)
- [Update documentation link in output](https://github.com/runtimeverification/kontrol/issues/58)

### The Week That Was (29 Sep - 5 Oct)
I will try to normalize this over the next few weeks, but for this week I want to focus on some general items that the team has been working on over the last several weeks to reorganize the work on Kontrol.

The Triage status was a bit of a dumping ground with several aged items, so the team spent a few Standup meetings moving these items to the Backlog, In Progress, etc. to empty the Triage list except for items added since the last Standup.

Next, we added a new section to the Project Board inspired by the K Team titled Next Up, which is a list of the highest priority items from the Backlog. When the team completes an item or finds themself with bandwidth, they will first look at Next Up to ensure that we are focused on the most important items.

We added a few other nuances to the project board, listed below:
- More priority classifications to add subtlety to the relative priority of items in the Backlog
- Classification field to group items into Issue categories
- Added Blocked, Investigation, and Maintenance statuses to improve Project Board organization

Once we reworked these elements, Palina and I worked on re-prioritizing the Backlog and reviewed the Backlog with the team to ensure that we got the order right. From here, we are using the following general criteria to assess the relative priority of new issues opened in Kontrol:
- If the Issue is critical for the success of a paid engagement, it goes to the top of the list
- If the Issue significantly improves the Performance of Kontrol, then it is the next highest priority
- If the Issue significantly improves the User Experience, then it is balanced against the Performance Improvements from the previous item
- Remaining items are prioritized below these three classifications as agreed upon by the team

It is important to note that, much like the Pirate Code, these are guidelines as opposed to hard and fast rules, meaning that each item can be considered higher or lower priority depending on a number of factors that are not addressed in the criteria above.

With this new framework in place, the team meets Tuesday and Thursday to work through the Triage list to avoid a buildup, then update status on Issues In Progress, and assign new tasks to team members with bandwidth. In weeks to come I will update this Issue in the Maintenance section of the Project Board with details on how the work is progressing, so look back here every Friday for the latest news on Kontrol.
